### PR TITLE
Reconfigure parameter for install

### PR DESCRIPTION
**Pull request by stephenlauck**
It would be  nice to be able to do a reconfigure without using notifies in order to control timing if you need an ingredient installed and reconfigured in one step.

### CODE REVIEW

File: libraries/chef_server_ingredients_resource.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ChefServerIngredient < Chef::Resource::LWRPBase
       attribute :master_token, :kind_of => String
       attribute :repository, :kind_of => String, :default => 'chef/stable'
       attribute :version, :kind_of => String, :default => nil
-      attribute :installed, :kind_of => [TrueClass, FalseClass, NilClass], :default => false
+      attribute :reconfigure, kind_of: [TrueClass, FalseClass], default: false
```
Comment (jtimberman) [inline libraries/chef_server_ingredients_resource.rb]: No hipster hashes! Hash rockets for life. :rocket: :smile:

File: libraries/chef_server_ingredients_provider.rb [hunk 2]
```diff
@@ -74,6 +81,13 @@ def ctl_command
         new_resource.ctl_command || chef_server_ctl_command(new_resource.package_name)
       end
 
+      def reconfigure
+        ctl_cmd = ctl_command
+        execute "#{new_resource.package_name}-reconfigure" do
+          command "#{ctl_cmd} reconfigure"
```
Comment (jtimberman) [inline libraries/chef_server_ingredients_provider.rb]: why does `ctl_cmd` need to be here?

Comment (jtimberman) [inline libraries/chef_server_ingredients_provider.rb]: Ugh, is the private method not available in the properties of a resource? :cry:

File: libraries/chef_server_ingredients_resource.rb [hunk 3]
```diff
@@ -16,7 +16,7 @@ class ChefServerIngredient < Chef::Resource::LWRPBase
       attribute :master_token, :kind_of => String
       attribute :repository, :kind_of => String, :default => 'chef/stable'
       attribute :version, :kind_of => String, :default => nil
-      attribute :installed, :kind_of => [TrueClass, FalseClass, NilClass], :default => false
```
Comment (jtimberman) [inline libraries/chef_server_ingredients_resource.rb]: We need this, it's a state attr for Chef Reporting.

**Comment by poliva83** [comment]
@stephenlauck These LWRPs seem to do similar job to the resource/providers created in the cheffish gem (except your using chef-server-ctl wrapper instead of using chef-server-api directly). I have had success using cheffish although the documentation is pretty much non-existent at the moment.

https://github.com/chef/cheffish
https://github.com/chef/cheffish/blob/master/lib/chef/resource/chef_user.rb
https://github.com/chef/cheffish/blob/master/lib/chef/resource/chef_organization.rb

File: libraries/chef_server_ingredients_provider.rb (hunk 2 above)
Comment (stephenlauck) [inline libraries/chef_server_ingredients_provider.rb]: I assumed that you had it right so I just replicated, but I'm not sure if you can use ctl_cmd directly there or not.

Comment (jtimberman) [inline libraries/chef_server_ingredients_provider.rb]: Yeah pretty sure this is correct on thinking about it.

**Pull request merged**